### PR TITLE
re-enable warnings for winml builds and fix the warnings that were hiding

### PR DESCRIPTION
**Pull request by martinb35**
we temporarily disabled "treat warnings as errors" to get some test code in. This change re-enables treating them as errors and fixes the warnings that were in the code.

Also clang-formatted protobufhelpers.cpp

### CODE REVIEW

**Comment by walrusmcd** [comment]
> #define _SILENCE_ALL_CXX17_DEPRECATION_WARNINGS

what does this one do?    

---
Refers to: winml/test/common/protobufHelpers.cpp:1 in d3b7770. [](commit_id = d3b77707adff7bb0de74cd9bef7a9928cdab8395, deletion_comment = False)

File: winml/test/scenario/cppwinrt/scenariotestscppwinrt.cpp [hunk 1]
```diff
@@ -1061,10 +1067,12 @@ TEST_F(ScenarioCppWinrtGpuTest, DISABLED_Scenario22_ImageBindingAsGPUTensor)
     );
 
     // Create the GPU upload buffer.
+    CD3DX12_HEAP_PROPERTIES props(D3D12_HEAP_TYPE_UPLOAD);
+    auto buffer = CD3DX12_RESOURCE_DESC::Buffer(bufferbytesize);
```
Comment (walrusmcd) [inline winml/test/scenario/cppwinrt/scenariotestscppwinrt.cpp]: this was a change for warnings?   or an unrelated change?

File: winml/test/scenario/cppwinrt/scenariotestscppwinrt.cpp [hunk 2]
```diff
@@ -1420,7 +1439,7 @@ TEST_F(ScenarioCppWinrtTest, EncryptedStream)
     LearningModel model = nullptr;
     EXPECT_NO_THROW(model = LearningModel::LoadFromStream(RandomAccessStreamReference::CreateFromStream(decryptedStream)));
     LearningModelSession session = nullptr;
-    EXPECT_NO_THROW(auto session = LearningModelSession(model));
```
Comment (walrusmcd) [inline winml/test/scenario/cppwinrt/scenariotestscppwinrt.cpp]: >auto [](start = 20, length = 4)

that's awesome.  we had 2 sessions :)

File: winml/test/scenario/cppwinrt/scenariotestscppwinrt.cpp (hunk 1 above)
Comment (martinb35) [reply winml/test/scenario/cppwinrt/scenariotestscppwinrt.cpp]: warning C4238: nonstandard extension used:
 class rvalue used as lvalue
[https://docs.microsoft.com/en-us/cpp/error-messages/compiler-warnings/compiler-warning-level-4-c4238?view=vs-2019](https://docs.microsoft.com/en-us/cpp/error-messages/compiler-warnings/compiler-warning-level-4-c4238?view=vs-2019)
For compatibility with previous versions of Visual C++, Microsoft extensions (/Ze) allow you to use a class type as an rvalue in a context that implicitly or explicitly takes its address. In some cases, such as the example below, this can be dangerous.

---
In reply to: [353328719](https://github.com/microsoft/onnxruntime/pull/2526#discussion_r353328719) [](ancestors = 353328719)

**Comment by martinb35** [comment]
> #define _SILENCE_ALL_CXX17_DEPRECATION_WARNINGS

we silence this error all over our project. At some point we should probably be more thoughtful about it. 

warning C4996: 'std::iterator<std::r
andom_access_iterator_tag,Element,ptrdiff_t,_Ty*,_Ty&>': warning STL4015: The std::iterator class template (used as a b
ase class to provide typedefs) is deprecated in C++17. (The <iterator> header is NOT deprecated.) The C++ Standard has
never required user-defined iterators to derive from std::iterator. To fix this warning, stop deriving from std::iterat
or and start providing publicly accessible typedefs named iterator_category, value_type, difference_type, pointer, and
reference. Note that value_type is required to be non-const, even for constant iterators. You can define _SILENCE_CXX17
_ITERATOR_BASE_CLASS_DEPRECATION_WARNING or _SILENCE_ALL_CXX17_DEPRECATION_WARNINGS to acknowledge that you have receiv
ed this warning. [B:\winmlpublic\build\Windows\Debug\winml_test_common.vcxproj]


---
In reply to: [561280000](https://github.com/microsoft/onnxruntime/pull/2526#issuecomment-561280000) [](ancestors = 561280000)

---
Refers to: winml/test/common/protobufHelpers.cpp:1 in d3b7770. [](commit_id = d3b77707adff7bb0de74cd9bef7a9928cdab8395, deletion_comment = False)
